OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -731,13 +731,20 @@ void TracerThread::Run(const std::shared_ptr<std::atomic<bool>>& exit_requested)
   }
 
   // Close the file descriptors.
-  {
+  // Do it in a separate dedicated thread as this can take a long time
+  // with a lot of uprobes and uretprobes.
+  std::thread([tracing_fds = tracing_fds_] {
+    pthread_setname_np(pthread_self(), "close(fd)s");
     ORBIT_SCOPE_WITH_COLOR("Closing file descriptors", orbit::Color::kRed);
-    for (int fd : tracing_fds_) {
+    uint64_t fd_close_begin_ns = MonotonicTimestampNs();
+    for (int fd : tracing_fds) {
       ORBIT_SCOPE("Closing fd");
       close(fd);
     }
-  }
+    uint64_t fd_close_end_ns = MonotonicTimestampNs();
+    LOG("Finished closing %lu perf_event_open file descriptors after %.3f ms", tracing_fds.size(),
+        (fd_close_end_ns - fd_close_begin_ns) / 1e6);
+  }).detach();
```

Comment:
I am not sure detach here is a good idea... maybe use thread_pool for this?

Reply:
We could use a static thread pool so that we would not spawn a new thread on every capture stop, but reuse the same one. If anything, it would mean that we would have a single introspection track in the UI instead of many.

Reply:
I mean, sure, I can just add a `static` thread pool as local variable. But apart from the number of introspection tracks, which I don't think should drive this decision, but is still something, I don't understand the arguments in favor of a thread pool. Thread pools are used for multiple and frequent short-lived tasks, and this is the exact opposite of it, with the cost of creating a new thread being irrelevant (and instead having an additional thread pool with one thread always alive).

Reply:
just to clarify why... detached thread is basically we do not know the scope of this thread - main thread can be dead at any moment which means some shared resources could become unreachable with its resources. TBH I do not even know if there is a valid case when detach should be used.

In other words someone need to manage this thread scope. Another approach is to remove the threading from here and delegate it to the caller via FileDescriptorCloser functor/interface if this helps to manage the scope better (not in this component).

Reply:
> main thread can be dead at any moment which means some shared resources could become unavailable.

What shared resources are we talking about in this specific case?

Reply:
log file descriptor?

Reply:
Great, so it seems that the only thing to do is to remove logging from here! This is a joke, but only half a joke.

It's only half a joke because this is supposed to be just a fix to the regression. Having a detached thread and no logging would then do exactly that. If we want to or need to properly handle synchronization and all, then at this point we should maybe just do it properly and communicate with the ui, at least we get something out of it.